src/app/highlight.directive.ts

```diff
@@ -1,45 +1,29 @@
-import { Directive , OnChanges, Input, SimpleChanges, Renderer2, ElementRef,} from '@angular/core';
+import { Directive , Input, ElementRef, Renderer2, OnChanges} from '@angular/core';
 
 @Directive({
   selector: '[appHighlight]'
 })
 export class HighlightDirective implements OnChanges{
 
-  @Input() searchedWord: string =''; // searchText
-  @Input() content: string = ''; // HTML content
-  @Input() classToApply: string = ''; //class to apply for highlighting
-  @Input() setTitle : boolean = false; //sets title attribute of HTML
-
+  @Input() appHighlight: string =''; // searchedWord
+  @Input() filtredList: string = ''; // filteredList
+  
   constructor(private el: ElementRef, private renderer: Renderer2) { }
+  
+  ngOnChanges(): void {
+    this.replaceWithHighlight(this.appHighlight.toLocaleLowerCase(), this.filtredList)
+  }
 
-  ngOnChanges(changes: SimpleChanges): void {
-
-    if (!this.content) {
-      return;
-    }
-
-    if (this.setTitle) {
-      this.renderer.setProperty(
-        this.el.nativeElement,
-        'title',
-        this.content
-      );
-    }
-
-    if (!this.searchedWord || !this.searchedWord.length || !this.classToApply) {
-      this.renderer.setProperty(this.el.nativeElement, 'innerHTML', this.content);
-      return;
+  replaceWithHighlight(searchWorld: string, list: string){
```

Comment:
```suggestion
  replaceWithHighlight(): void {
```
you have the variables globally, you can access them directly in the method